pkg/state/state_test.go

```diff
@@ -56,6 +59,70 @@ func TestSchemaOptionIsRespected(t *testing.T) {
 	})
 }
 
+func TestInferredMigration(t *testing.T) {
+	t.Parallel()
+
+	testutils.WithStateAndConnectionToContainer(t, func(state *state.State, db *sql.DB) {
+		ctx := context.Background()
+
+		tests := []struct {
+			name          string
+			sqlStmt       string
+			wantMigration migrations.Migration
+		}{
+			{
+				name:    "create table",
+				sqlStmt: "CREATE TABLE public.table1 (id int)",
+				wantMigration: migrations.Migration{
+					Operations: migrations.Operations{
+						&migrations.OpRawSQL{
+							Up: "CREATE TABLE public.table1 (id int)",
+						},
+					},
+				},
+			},
+		}
+
+		// init the state
+		if err := state.Init(ctx); err != nil {
+			t.Fatal(err)
+		}
```

Comment:
I think this state initalization should be moved inside the `WithStateAndConnectionToContainer` helper. The other usages of this helper all init the state outside too.